microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -128,9 +129,16 @@ func createService(step string, ns string, svc string, svcCfg *map[string]string
 
 		gvr, _ := meta.UnsafeGuessKindToResource(*gvk)
 		for {
-			// _, err = dynamicClient.Resource(gvr).Namespace(ns).List(context.TODO(), metav1.ListOptions{})
+			patchBytes, err := json.Marshal(obj)
+			if err != nil {
+				return fmt.Errorf("failed to marshal config to json: %v", err)
+			}
 
-			createdObj, err := dynamicClient.Resource(gvr).Namespace(ns).Create(context.TODO(), obj, metav1.CreateOptions{})
+			// createdObj, err := dynamicClient.Resource(gvr).Namespace(ns).Create(context.TODO(), obj, metav1.CreateOptions{})
```

Comment:
TODO usage in English is like a message to the developer to complete something. any better word than context.TODO or is this some coding convention?

Reply:
It's a placeholder for when the correct context to use is not known. It is not meant to be used indefinitely. The idea is to use context.TODO()